packages/teams-js/src/public/runtime.ts

```diff
@@ -127,6 +159,61 @@ export const v1HostClientTypes = [
   HostClientType.teamsDisplays,
 ];
 
+interface IRuntimeUpgrade {
+  versionToUpgradeFrom: number;
+  upgradeToNextVersion: (previousVersionRuntime: IBaseRuntime) => IBaseRuntime;
+}
+
+/**
+ * @internal
+ * Limited to Microsoft-internal use
+ *
+ * Uses upgradeChain to transform an outdated runtime object to the latest format.
+ * @param outdatedRuntime - The runtime object to be upgraded
+ * @returns The upgraded runtime object
+ * @throws Error if the runtime object could not be upgraded to the latest version
+ */
+function fastForwardRuntime(outdatedRuntime: IBaseRuntime): Runtime {
+  let runtime = outdatedRuntime;
+  if (runtime.apiVersion < latestRuntimeApiVersion) {
+    upgradeChain.forEach((upgrade) => {
+      if (runtime.apiVersion === upgrade.versionToUpgradeFrom) {
+        runtime = upgrade.upgradeToNextVersion(runtime);
+      }
+    });
+  }
+  if (isLatestRuntimeVersion(runtime)) {
```

Comment:
Might be good to open a task on our backlog for handling truly breaking runtime changes at some point